src/scripts/core/models/column.ts

```diff
@@ -14,24 +15,37 @@ export class Column extends Container{
         this.columnID = columnID;
         this.columnSize = columnSize;
         this.availHeight = availHeight;
-        for(var i = 0; i < this.columnSize; i++) {
-            const newToken = new Token(this.columnID, i, this.columnSize, availWidth, availHeight);
-            newToken.y = (availHeight / (this.columnSize/i));
+        for(var tokenIndex: number = 0; tokenIndex < this.columnSize; tokenIndex++) {
```

Comment:
Just a note, we tend to use 'let' and not 'var' when declaring a variable

Reply:
To explain the reasoning behind this as well. When you have a 'var' declared, javascript 'hoists' these values when compiling. Functionally declaring them at the top of the scope they are found in (In this case the function). This means that if you declare a var within a for loop it is actually available outside of that for loop.

'let' and 'const' are not hoisted and only available where they are declared. In addition if you declare a let/const in an if statement or in a for loop for example, it will not be available outside of that if/for. A good way to know where a let/const will be available is to identify which '{ }' brackets they are within.

Incidentally, there are a number of 'var' declarations like this that should be 'let' or 'const'. I recommend doing a pass to find these